### ISSUE TITLE
Bug: "list assignment index out of range"

### ISSUE THREAD

**Issue by selone** [nasa/fmdtools#22]
### Environment Information
OS: linux Centos8
python: cpython 3.9.16

### Triggering Action
When I run a pump case
```
>>> import fmdtools.sim.propagate as propagate
>>> from fmdtools.sim.sample import FaultDomain, FaultSample
>>> from fmdtools.analyze.phases import PhaseMap
>>> import fmdtools.analyze as an
>>> from ex_pump import Pump
>>> mdl = Pump()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "XXXXXX/fmdtools/examples/pump/ex_pump.py", line 412, in __init__
    super().__init__(**kwargs)
  File "XXXXXX/fmdtools/fmdtools/define/model.py", line 49, in __init__
    super().__init__(name=name, p=p, sp=sp, r=r, track=track)
  File "XXXXXX/fmdtools/fmdtools/define/block.py", line 136, in __init__
    init_obj_attr(self, p=p, sp=sp, r=r)
  File "XXXXXX/fmdtools/fmdtools/define/common.py", line 297, in init_obj_attr
    setattr(obj, at, init_at(**at_arg))
  File "XXXXXX/fmdtools/fmdtools/define/rand.py", line 61, in __init__
    args = get_true_fields(self, *args, seed=seed,
  File "XXXXXX/fmdtools/fmdtools/define/common.py", line 212, in get_true_fields
    true_args[i] = kwargs[n]
IndexError: list assignment index out of range
```

**Comment by hulsed** [comment]
@selone could you list your version of `recordclass`? I suspect this is an issue of [dataclass](https://pypi.org/project/recordclass/) changing between versions

**Comment by hulsed** [comment]
This is now fixed in v2.0-beta3
